### PR TITLE
PXP-4102 Feat/arb admin

### PR DESCRIPTION
**Pull request by vpsx**
Switches Sheepdog authz handling of program/project CRUD to centralized auth/Arborist.

See https://github.com/uc-cdis/commons-users/pull/810 for new Sheepdog admin policy.

(For QAs' benefit) How to test:
- See PR linked above for example of how to toggle for a given user the old `admin: true/false` and the new admin policy 
- Verify that the API behaves according to this matrix:
```
    User DOES have Sheepdog admin policy: All CRUD authorized.
        Admin is FALSE:
            Create program: Authorized
            Delete program: Authorized
            Create project: Authorized
            Delete project: Authorized
        Admin is TRUE:
            Create program: Authorized
            Delete program: Authorized
            Create project: Authorized
            Delete project: Authorized
    User does NOT have Sheepdog admin policy: All CRUD not authorized.
    (Also check that they are 403s in particular)
        Admin is FALSE:
            Create program: Unauthorized
            Delete program: Unauthorized
            Create project: Unauthorized
            Delete project: Unauthorized
        Admin is TRUE:
            Create program: Unauthorized
            Delete program: Unauthorized
            Create project: Unauthorized
            Delete project: Unauthorized
```

### New Features
Switch checks guarding program/project CRUD to centralized auth (instead of checking for "admin" in JWT, check for Sheepdog admin policy in Arborist)

### Breaking Changes
Having admin: true in user.yaml no longer allows a user to do program/project CRUD in Sheepdog. Users now need to have the Sheepdog admin policy in Arborist. 

### Deployment changes
The environment's user.yaml needs to be updated to add the new Sheepdog admin policy/resources/roles; for each user that has admin: true, grant them the new policy. See https://github.com/uc-cdis/commons-users/pull/810 for an example.

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
This PR contains code that is not formatted correctly according to [`black`](https://github.com/ambv/black). Run `black` on your code before merging, or comment "wool" or "black" on this PR and I will format the code for you.

<details>
<summary>Expand the full diff to see formatting changes</summary>

```diff
--- sheepdog/auth/__init__.py
+++ blackened
@@ -69,18 +69,19 @@
 def require_sheepdog_program_admin(func):
     """
     Wrap a function to allow access to the handler if the user has access to
     the resource /services/sheepdog/submission/program (Sheepdog program admin)
     """
+
     @functools.wraps(func)
     def authorize_and_call(*args, **kwargs):
         jwt = get_jwt_from_header()
         authz = flask.current_app.auth.auth_request(
-            jwt = jwt,
-            service = "sheepdog",
-            methods = "*",
-            resources = ["/services/sheepdog/submission/program"],
+            jwt=jwt,
+            service="sheepdog",
+            methods="*",
+            resources=["/services/sheepdog/submission/program"],
         )
         if not authz:
             raise AuthZError("Unauthorized: User must be Sheepdog program admin")
         return func(*args, **kwargs)
 
@@ -90,18 +91,19 @@
 def require_sheepdog_project_admin(func):
     """
     Wrap a function to allow access to the handler if the user has access to
     the resource /services/sheepdog/submission/project (Sheepdog project admin)
     """
+
     @functools.wraps(func)
     def authorize_and_call(*args, **kwargs):
         jwt = get_jwt_from_header()
         authz = flask.current_app.auth.auth_request(
-            jwt = jwt,
-            service = "sheepdog",
-            methods = "*",
-            resources = ["/services/sheepdog/submission/project"],
+            jwt=jwt,
+            service="sheepdog",
+            methods="*",
+            resources=["/services/sheepdog/submission/project"],
         )
         if not authz:
             raise AuthZError("Unauthorized: User must be Sheepdog project admin")
         return func(*args, **kwargs)
 
```
</details></br>

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

File: tests/integration/datadict/submission/test_admin_endpoints.py [hunk 1]
```diff
@@ -94,63 +94,61 @@ def test_to_delete(
...
+    client, pg_driver, cgci_blgsp, index_client, submitter, require_index_exists_off
+):
+    """Try to reassign a node's remote URL
 
-    Returns:
-        requests.Response: http response from doing reassign request
-        string: did
-        string: s3 url that you changed it to
+    Url:
+        PUT: /admin/<program>/<project>/files/<file_uuid>/reassign
```
Comment (paulineribeyre) [inline tests/integration/datadict/submission/test_admin_endpoints.py]: this (updating URLs) should be supported in indexd, not sheepdog... and this code only supports S3 URLs, and it assumes there is only 1 stored URL 🤦‍♀ we should probably get rid of this and add an endpoint in indexd instead

cc @Avantol13 re:conversation about indexd not supporting URL updates

**Comment by vpsx** [comment]
Thanku!! I completely forgot 🤦‍♂ . So that's all done except I didn't add the warning to the other tests--they didn't really claim to be checking authz anyway (whereas the first few either used to check authz, or were next to tests that checked the unauthorized case, so there was the implication). And it would have been a lot of noise :/

File: tests/integration/datadict/submission/test_admin_endpoints.py (hunk 1 above)
Comment (vpsx) [reply tests/integration/datadict/submission/test_admin_endpoints.py]: Should we wait until indexd has the endpoint or shall I go ahead and remove this~

Comment (Avantol13) [reply tests/integration/datadict/submission/test_admin_endpoints.py]: we can wait to be safe, though I don't think it's actually used anywhere? but if we change the public API we need a new major version of sheepdog

**Pull request merged**